File: 0575-distribute-candies/0575-distribute-candies.java
class Solution {
    public int distributeCandies(int[] candies) {
        Set<Integer> set = new HashSet<>();
        for (int i : candies) {
            set.add(i);
        }
        int size = set.size();
        int n = candies.length;
        return Math.min(size, n / 2);
    }
}

File: 0575-distribute-candies/0575-distribute-candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candies) {
         unordered_set<int>s;
   int n=candies.size();
   for(int i:candies)s.insert(i);
   int size=s.size();
   return min(size,n/2);
    }
};